Framework/DataHandling/src/LoadMcStasNexus.cpp

```diff
@@ -42,10 +43,19 @@ const std::string LoadMcStasNexus::category() const { return "DataHandling\\Nexu
  * @returns An integer specifying the confidence level. 0 indicates it will not
  * be used
  */
-int LoadMcStasNexus::confidence(Kernel::NexusDescriptor &descriptor) const {
-  UNUSED_ARG(descriptor)
-  // To ensure that this loader is somewhat hitten return 0
+int LoadMcStasNexus::confidence(Kernel::NexusHDF5Descriptor &descriptor) const {
   int confidence(0);
+  const auto entries = descriptor.getAllEntries();
+  for (const auto &[nx_class, grouped_entries] : entries) {
+    UNUSED_ARG(nx_class);
+    for (const auto &path : grouped_entries) {
+      // Mccode writes an information dataset so can be reasonably confident if we find it
+      if (std::regex_search(path, std::regex("information$"))) {
```

Comment:
The performance of `std::regex` is known to be poor. Is this just an "ends with" check? Could we use [boost::ends_with](https://www.boost.org/doc/libs/1_60_0/doc/html/boost/algorithm/ends_with.html) check instead?